app/uk/gov/hmrc/tai/model/CodingComponentPairModel.scala

```diff
@@ -32,12 +33,13 @@ object CodingComponentPairModel {
 
   def apply(codingComponentPair: CodingComponentPair,
             employmentIds: Map[Int, String],
-            companyCarBenefits: Seq[CompanyCarBenefit])
+            companyCarBenefits: Seq[CompanyCarBenefit],
+            totalTax: TotalTax)
            (implicit messages: Messages): CodingComponentPairModel = {
```

Comment:
The issue here with passing so many things around is that theres no description for this blob of data, if you wrap all of these 4 arguments up in to a case class (not sure what to call it yet) you can give the totalTax, and other params, a default value. This is called the parameter object pattern (https://refactoring.guru/introduce-parameter-object). It should shorten up your tests a lot, especially the view model ones.